src/app/SecurityHelper.ts

```diff
@@ -9,9 +9,10 @@ const scope = encodeURIComponent(oAuthScope);
 
 /// generates and saves an oauth state to sessionStorage (oauth-state) and returns url of discord oauth
 export function navigateToDiscordOauth() {
-  // set oauth state in session storage
+  // set oauth state in local storage
+  // Discord OAuth2 opens a new tab on Android Chrome, so we use localStorage instead of sessionStorage
   const state = generateOauthState();
-  sessionStorage.setItem('oauth-state', state);
+  localStorage.setItem('expected-oauth-state', state);
```

Comment:
You might want to wrap local storage accessors in some util that will try catch around them and ensure the calling logic can handling missing data. Depending on browser, user settings, private browsing mode, these can error even on supporting browsers.

Here's the utils I use that include comments detailing the reasons they may error:
https://github.com/DnDBeyond/ddb-character-listing-client/blob/master/src/LocalStorageUtils.js